### DISCUSSION TITLE
COG notebook exporting geotif from qgis

### DISCUSSION THREAD

**Discussion by Niko-La** [opengeos/leafmap#173] [Q&A] [answered]
1)is there a special way to export geotiff from qgis?
2)possible to host geotif fille on github and dsiplay?

![image](https://user-images.githubusercontent.com/14004559/148899272-c9f011f7-2c25-41c0-902d-bd394010827b.png)

**Comment by giswqs** [comment, marked as answer]
You need to covert the image to COG. See https://github.com/banesullivan/localtileserver/discussions/27

Once you have the COG, you can host it on GitHub and use leafmap to load COG onto the map. 

```python
m = leafmap.Map()
url = "https://github.com/giswqs/data/raw/main/raster/cog.tif"
m.add_cog_layer(url, name="COG", bands=[4, 1, 2])
m
```
![image](https://user-images.githubusercontent.com/5016453/149025274-e6a0e07e-2392-4d66-bddd-75b825a25770.png)

**Comment by giswqs** [comment]
See https://leafmap.org/notebooks/42_create_cog

**Comment by Niko-La** [comment]
Thansk  notebook demo works great on your geotiff file.  If there is a format error when validating, is there a specific way to export grom qgis to have the right version?

`---------------------------------------------------------------------------
CPLE_OpenFailedError                      Traceback (most recent call last)
rasterio/_base.pyx in rasterio._base.DatasetBase.__init__()

rasterio/_shim.pyx in rasterio._shim.open_dataset()

rasterio/_err.pyx in rasterio._err.exc_wrap_pointer()

CPLE_OpenFailedError: '/vsicurl/https://github.com/Niko-La/jupyter_equity_2020/blob/main/brampton_geo.tif' not recognized as a supported file format.

During handling of the above exception, another exception occurred:

RasterioIOError                           Traceback (most recent call last)
<ipython-input-7-8dd92386fca4> in <module>()
      1 #!pip install rio-cogeo
----> 2 leafmap.cog_validate(url2)

3 frames
/usr/local/lib/python3.7/dist-packages/rasterio/__init__.py in open(fp, mode, driver, width, height, count, crs, transform, dtype, nodata, sharing, **kwargs)
    218         # None.
    219         if mode == 'r':
--> 220             s = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)
    221         elif mode == "r+":
    222             s = get_writer_for_path(path, driver=driver)(

rasterio/_base.pyx in rasterio._base.DatasetBase.__init__()

RasterioIOError: '/vsicurl/https://github.com/Niko-La/jupyter_equity_2020/blob/main/brampton_geo.tif' not recognized as a supported file format.`

**Reply by giswqs** [reply]
The URL should be https://github.com/Niko-La/jupyter_equity_2020/raw/main/brampton_geo.tif.
